h/storage.py

```diff
@@ -217,37 +218,54 @@ def update_annotation(request, id_, data, group_service):
     return annotation
 
 
-def expand_uri(session, uri):
+def expand_uri(session, uri, normalized=False):
     """
     Return all URIs which refer to the same underlying document as `uri`.
 
     This function determines whether we already have "document" records for the
     passed URI, and if so returns the set of all URIs which we currently
     believe refer to the same document.
 
-    :param session: the database session
-    :type session: sqlalchemy.orm.session.Session
-
-    :param uri: a URI associated with the document
-    :type uri: str
+    :param session: Database session
+    :param uri: URI associated with the document
+    :param normalized: Return normalized URIs instead of the raw value
 
     :returns: a list of equivalent URIs
-    :rtype: list
     """
-    doc = models.Document.find_by_uris(session, [uri]).one_or_none()
 
-    if doc is None:
-        return [uri]
+    canonical_url = normalize_uri(uri)
```

Comment:
It's a super tiny saving, but by pulling this up here we only ever run it once. In the previous version, `find_by_uris` would do it once, and we'd do it again in the failure cases